lib/modules/asset/AssetsGroupsController.ts

```diff
@@ -0,0 +1,239 @@
+          ],
+        },
+        update: {
+          handler: this.update.bind(this),
+          http: [
+            { path: "device-manager/:engineId/assetsGroups/:_id", verb: "put" },
+          ],
+        },
+        delete: {
+          handler: this.delete.bind(this),
+          http: [
+            {
+              path: "device-manager/:engineId/assetsGroups/:_id",
+              verb: "delete",
+            },
+          ],
+        },
+        search: {
+          handler: this.search.bind(this),
+          http: [
+            {
+              path: "device-manager/:engineId/assetsGroups/_search",
+              verb: "get",
+            },
+            {
+              path: "device-manager/:engineId/assetsGroups/_search",
+              verb: "post",
+            },
+          ],
+        },
+        addAsset: {
+          handler: this.addAsset.bind(this),
+          http: [
+            {
+              path: "device-manager/:engineId/assetsGroups/:_id/addAsset",
+              verb: "post",
+            },
+          ],
+        },
+      },
+    };
+    /* eslint-enable sort-keys */
+  }
+
+  private get sdk() {
+    return this.plugin.context.accessors.sdk;
+  }
+
+  private get as() {
+    return (user: User | null): EmbeddedSDK => {
+      if (user?._id) {
+        return this.sdk.as(user, { checkRights: true });
+      }
+      return this.sdk;
+    };
+  }
+
+  async checkParent(
+    engineId: string,
+    body: AssetsGroupsBodyRequest
+  ): Promise<void> {
+    if (typeof body.parent === "string") {
+      const parentExist = await this.sdk.document.exists(
+        engineId,
+        InternalCollection.ASSETS_GROUPS,
+        body.parent
+      );
+      if (!parentExist) {
+        throw new BadRequestError(
+          `The parent group "${body.parent}" does not exist`
+        );
+      }
+    }
+  }
+
+  async checkChildren(
+    engineId: string,
+    body: AssetsGroupsBodyRequest
+  ): Promise<void> {
+    if (Array.isArray(body.children)) {
+      for (const childrenId of body.children) {
+        const childrenExist = await this.sdk.document.exists(
+          engineId,
+          InternalCollection.ASSETS_GROUPS,
+          childrenId
+        );
+        if (!childrenExist) {
+          throw new BadRequestError(
+            `The children group "${childrenId}" does not exist`
+          );
+        }
+      }
+    }
+  }
+
+  async create(request: KuzzleRequest): Promise<ApiGroupCreateResult> {
+    const engineId = request.getString("engineId");
+    const _id = request.getId({ ifMissing: "error" });
+    const body = request.getBody() as AssetsGroupsBodyRequest;
+
+    await this.checkParent(engineId, body);
+
+    return this.as(request.getUser()).document.create<AssetsGroupsBody>(
+      engineId,
+      InternalCollection.ASSETS_GROUPS,
+      {
+        children: [],
+        name: body.name,
+        parent: body.parent ?? null,
```

Comment:
body.parent ?? null is not equals to just do body.parent

Reply:
Because `body.parent` could be undefined, so `body.parent ?? null` define null like default value